hudi-common/src/main/java/org/apache/hudi/common/util/collection/ExternalSpillableMap.java

```diff
@@ -221,7 +221,7 @@ public R put(T key, R value) {
       // At first, use the sizeEstimate of a record being inserted into the spillable map.
       // Note, the converter may over-estimate the size of a record in the JVM
       this.estimatedPayloadSize = keySizeEstimator.sizeEstimate(key) + valueSizeEstimator.sizeEstimate(value);
-    } else if (this.inMemoryMap.size() % NUMBER_OF_RECORDS_TO_ESTIMATE_PAYLOAD_SIZE == 0) {
+    } else if (this.currentInMemoryMapSize < this.maxInMemorySizeInBytes && this.inMemoryMap.size() % NUMBER_OF_RECORDS_TO_ESTIMATE_PAYLOAD_SIZE == 0) {
```

Comment:
so when the memory threshold reaches, the size will be a constant?

Reply:
right, since the new entries all go to the disk map

Reply:
Can we move the whole size estimation into the if clause 234 ~ 241, because only that part needs the `estimatedPayloadSize` to accumulate with the existing `currentInMemoryMapSize`.

Reply:
If you do an update and the key is already in the in-memory map, the size is not recalculated in the current implementation.